core/src/listener/listener.ts

```diff
@@ -215,30 +220,41 @@ function latestJob({
       throw e
     }
 
+    // happens only on localhost
     if (latestBlock < observedBlock) {
       logger.warn('latestBlock < observedBlock. Updating observed block to revert the condition.')
       observedBlock = Math.max(0, latestBlock - 1)
     }
 
-    const logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
     try {
+      const logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
```

Comment:
it is defined twice, here in line 230 and also at line 268, with the same parameter. it'd be okay to rollback taking out from the try block

Reply:
They have different parameters :)

Reply:
awh so are `contractAddress, observedBlock, latestBlock` having different values?
```
const logPrefix = generateListenerLogPrefix(contractAddress, observedBlock, latestBlock)
```

Reply:
ah the second definition (in case of an exception) should have been from `latestObservedBlock` to `latestBlock`. I'll update it, thanks for catching!